src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -11760,3 +11931,435 @@ pbsv1mod_meth_resource_str_value(PyObject *self, PyObject *args, PyObject *kwds)
+		log_err(-1, __func__, "strdup exec_vnode failed");
+		goto release_nodes_exit;
+	}
+
+	/* exec_host or exec_host2 */
+	tmpstr = pbs_python_object_get_attr_string_value(py_job, ATTR_exechost);
+	if ((tmpstr != NULL) && (tmpstr[0] != '\0')) {
+		exechost = strdup(tmpstr);
+		if (exechost == NULL) {
+			log_err(-1, __func__, "strdup exec_host failed");
+			goto release_nodes_exit;
+		}
+	}
+
+	tmpstr = pbs_python_object_get_attr_string_value(py_job, ATTR_exechost2);
+	if ((tmpstr != NULL) && (tmpstr[0] != '\0')) {
+		exechost2 = strdup(tmpstr);
+		if (exechost2 == NULL) {
+			log_err(-1, __func__, "strdup exec_host2 failed");
+			goto release_nodes_exit;
+		}
+	}
+
+	if ((exechost == NULL) && (exechost2 == NULL)) {
+		log_err(-1, __func__, "no value found for exec_host/exec_host2 ");
+		goto release_nodes_exit;
+	}
+
+	if (exechost == NULL) {
+		exechost = strdup(exechost2);
+		if (exechost == NULL) {
+			log_err(-1, __func__, "strdup exec_host failed");
+			goto release_nodes_exit;
+		}
+	}
+
+	if (exechost2 == NULL) {
+		exechost2 = strdup(exechost);
+		if (exechost2 == NULL) {
+			log_err(-1, __func__, "strdup exec_host2 failed");
+			goto release_nodes_exit;
+		}
+	}
+
+	/* schedselect */
+	tmpstr = pbs_python_object_get_attr_string_value(py_job, ATTR_SchedSelect);
+	if ((tmpstr == NULL) || (tmpstr[0] == '\0')) {
+		log_err(-1, __func__, "did not find a value to exec_vnode");
+		goto release_nodes_exit;
+	}
+	schedselect = strdup(tmpstr);
+	if (schedselect == NULL) {
+		log_err(-1, __func__, "strdup schedselect failed");
+		goto release_nodes_exit;
+	}
+
+	/* populate exec_vnode_list */
+	if (populate_svrattrl_from_vnodelist_param(PY_EVENT_PARAM_VNODELIST, &exec_vnode_list) != 0) {
+		snprintf(log_buffer, sizeof(log_buffer), "%s: Failed to dump Python string list values into a svrattrl list", PY_EVENT_PARAM_VNODELIST);
+		log_err(PBSE_INTERNAL, __func__, log_buffer);
+		goto release_nodes_exit;
+	}
+
+	err_msg[0] = '\0';
+	relnodes_input_init(&r_input);
+	r_input.jobid = jobid;
+	r_input.vnodes_data = &exec_vnode_list;
+	r_input.execvnode = execvnode;
+	r_input.exechost = exechost;
+	r_input.exechost2 = exechost2;
+	r_input.p_new_exec_vnode = &new_exec_vnode;
+	r_input.p_new_exec_host = &new_exec_host;
+	r_input.p_new_exec_host2 = &new_exec_host2;
+	r_input.p_new_schedselect = &new_schedselect;
+
+	if (vnodelist != NULL) {
+		relnodes_input_vnodelist_init(&r_input_vnlist);
+		r_input_vnlist.vnodelist = vnodelist;
+		r_input_vnlist.schedselect = schedselect;
+		rc = pbs_release_nodes_given_nodelist(&r_input, &r_input_vnlist, err_msg, LOG_BUF_SIZE);
+		snprintf(log_buffer, sizeof(log_buffer), "release_nodes_given_nodelist: AFT rc=%d jobid=%s vnodelist=%s execvnode=%s exechost=%s exechost2=%s schedselect=%s new_exec_vnode=%s new_exec_host=%s new_exec_host2=%s new_schedselect=%s", rc, jobid, vnodelist, execvnode, exechost?exechost:"null", exechost2?exechost2:"null", schedselect, new_exec_vnode, new_exec_host, new_exec_host2, new_schedselect);
```

Comment:
I'd be surprised if gcc 8 doesn't complain about this. We should use pbs_asprintf() once the other PR gets merged.